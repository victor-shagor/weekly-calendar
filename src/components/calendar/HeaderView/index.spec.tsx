import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import Header from ".";

describe("Header view", () => {
  it("should render calendar header view", () => {
    render(<Header />);
    const btnDiv = screen.getByTestId("header");

    expect(btnDiv.querySelector(".btn")).toBeInTheDocument();
    expect(btnDiv.querySelectorAll(".btn").length).toEqual(2);
  });

  it("renders correctly", () => {
    const tree = renderer.create(<Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
