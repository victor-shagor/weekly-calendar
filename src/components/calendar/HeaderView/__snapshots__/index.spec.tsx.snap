// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header view renders correctly 1`] = `
<div
  className="calendar-header"
  data-testid="header"
>
  <h2 />
  <div>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height={25}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1"
        viewBox="0 0 48 48"
        width={25}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#2196F3"
          points="30.9,43 34,39.9 18.1,24 34,8.1 30.9,5 12,24"
        />
      </svg>
    </button>
    <button
      className="btn"
      onClick={[Function]}
      type="button"
    >
      <svg
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height={25}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1"
        viewBox="0 0 48 48"
        width={25}
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#2196F3"
          points="17.1,5 14,8.1 29.9,24 14,39.9 17.1,43 36,24"
        />
      </svg>
    </button>
  </div>
</div>
`;
